neural_compressor/torch/algorithms/static_quant/utility.py

```diff
@@ -63,57 +66,238 @@
+                        if (
+                            "op_type" in ipex_op_cfg
+                            and ipex_op_cfg["op_type"] == "<class 'torch.nn.modules.linear.Linear'>"
+                        ):
+                            smooth_quant_enable = True
+                        else:
+                            smooth_quant_enable = False
+                        activation_observer = generate_activation_observer(
+                            inc_scheme, inc_algorithm, smooth_quant, smooth_quant_enable
+                        )
+                        if not smooth_quant:
+                            if inc_scheme == "sym":
+                                input_tensor_infos[index]["force_dtype"] = "torch.qint8"
+                            if inc_scheme == "asym":
+                                input_tensor_infos[index]["force_dtype"] = "torch.quint8"
+                        ipex_op_cfg["activation_observer"] = activation_observer
+                    # int8 -> fp32
+                    else:
+                        input_tensor_infos[index]["force_dtype"] = "torch.float32"
+                    # modify pre_op output inf_dtype
+                    if i == 0:
+                        input_tensor_id = input_tensor_info["id"]
+                        input_tensor_dtype = input_tensor_info["force_dtype"]
+                        if input_tensor_id in output_tensor_ids_op_name.keys():
+                            pre_op_name = output_tensor_ids_op_name[input_tensor_id]
+                            pre_op_module = pre_op_name[0][0]
+                            pre_op_state = pre_op_name[0][1]
+                            pre_op_index = pre_op_name[0][2]
+                            pre_op_infos = cfgs[pre_op_module][pre_op_state][pre_op_index]
+                            pre_op_output_infos = pre_op_infos["output_tensor_infos"]
+                            for index, pre_op_output in enumerate(pre_op_output_infos):
+                                if pre_op_output["id"] == input_tensor_id:
+                                    pre_op_output_infos[index]["inf_dtype"] = input_tensor_dtype
+                                else:
+                                    pass
+                            pre_op_infos["output_tensor_infos"] = pre_op_output_infos
+                            cfgs[pre_op_module][pre_op_state][pre_op_index] = pre_op_infos
+                        else:
+                            pass
+            cfgs[name[0]][name[1]][name[2]] = ipex_op_cfg
+    return cfgs
+
+
+def generate_activation_observer(scheme, algorithm, smooth_quant=False, smooth_quant_enable=False):  # pragma: no cover
+    """This is a helper method to generate an activation observer.
+
+    Args:
+        scheme (str): Quantization scheme to be used.
+        algorithm (str): What algorithm for computing the quantization parameters based on.
+
+    Returns:
+        An observer.
+    """
+    kl_activation_observer = {

```

Comment:
Do these observer configurations already exist in IPEX? If so, kindly import them straight.